keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -0,0 +1,733 @@
+	
+Upon success, the `BucketAccess` instance is ready and the app workload can access backend storage.
+
+### Static Provisioning
+"Driverless" allows the existing workloads to make use of COSI without the need for vendors to create drivers. The following steps show the details of the workflow:
++ Admin creates `Bucket` instance that references an existing backend bucket.
++ User creates a `BucketRequest` (BR) that references this `Bucket`.
++ User creates `BucketAccessRequest` (BAR) that references their BR, a `BucketAccessClass`, and their secret containing access tokens.
++ COSI detects the existence of the BAR and follows the [grant access](#grant-bucket-access) steps.
++ COSI detects the existence of the BAR, BA and executes a validation process.
++ As a validation step, COSI ensures that the `BucketRequest.bucketInstanceName` matches the `BucketAccessRequest.bucketAccessName.bucketInstanceName`. In other words, the two related `Bucket` references match, meaning that the BR and BAR->BA point to the same `Bucket`.
+
+---
+
+# gRPC Definitions
+
+There is one service defined by COSI - `Provisioner`. This will need to be satisfied by the vendor-provisioner in order to be COSI-compatible
+
+```
+service Provisioner {
+    rpc ProvisionerGetInfo (ProvisionerGetInfoRequest) returns (ProvisionerGetInfoResponse) {}
+
+    rpc ProvisionerCreateBucket (ProvisionerCreateBucketRequest) returns (ProvisionerCreateBucketResponse) {}
+    rpc ProvisionerDeleteBucket (ProvisionerDeleteBucketRequest) returns (ProvisionerDeleteBucketResponse) {}
+
+    rpc ProvisionerGrantBucketAccess (ProvisionerGrantBucketAccessRequest) returns (ProvisionerGrantBucketAccessResponse);
+    rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
+}
+```
+
+## ProvisionerGetInfo
+
+This call is meant to retrieve the unique provisioner Identity. This identity will have to be set in `BucketRequest.Provisioner` field in order to invoke this specific provisioner.
+
+```
+message ProvisionerGetInfoRequest {
+    // Intentionally left blank
+}
+
+message ProvisionerGetInfoResponse {
+  // The name MUST follow domain name notation format
+  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
+  // include the plugin's host company name and the plugin name,
+  // to minimize the possibility of collisions. It MUST be 63
+  // characters or less, beginning and ending with an alphanumeric
+  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+  // alphanumerics between. This field is REQUIRED.
+  string name = 1;
+}
+```
+
+## ProvisonerCreateBucket
+
+This call is made to create the bucket in the backend. If a bucket that matches both name and parameters already exists, then OK (success) must be returned. If a bucket by same name, but different parameters is provided, then the appropriate error code `ALREADY_EXISTS` must be returned by the provisioner.
```

Comment:
Please add "ProvisonerCreateBucket MUST be idempotent".

Reply:
Fixed